test/components/views/messages/DateSeparator-test.tsx

```diff
@@ -114,16 +124,163 @@ describe("DateSeparator", () => {
     });
 
     describe("when feature_jump_to_date is enabled", () => {
+        const currentDate = new Date("2023-03-05");
         beforeEach(() => {
+            jest.clearAllMocks();
             mocked(SettingsStore).getValue.mockImplementation((arg): any => {
                 if (arg === "feature_jump_to_date") {
                     return true;
                 }
             });
+            jest.spyOn(dispatcher, "dispatch").mockImplementation(() => {});
         });
+
+        afterEach(async () => {
+            await clearAllModals();
```

Comment:
Created the `clearAllModals` utility so that we wouldn't see modals leak from test to test. Also noticed we doing some modal tending in other tests so instead of spreading the hacks further, I've consolidated them all to `clearAllModals` which we can strive to clean-up later.

Running [`cleanup()`](https://testing-library.com/docs/react-testing-library/api/#cleanup) from the React testing library, etc has no effect on cleaning up these modals.